Doc/Bib/MSC/06.html

```diff
@@ -125,6 +115,16 @@
 </p>
 
 
+<p class='BibEntry'>
+[<span class='BibKeyLink'><a href="http://www.ams.org/mathscinet-getitem?mr=2191285">J05</a></span>]   <b class='BibAuthor'>Jonsson, J.</b>,
+ <i class='BibTitle'>Simplicial complexes of graphs and hypergraphs with a bounded covering number</i>,
+ <span class='BibJournal'>SIAM J. Discrete Math.</span>,
+ <em class='BibVolume'>19</em> (<span class='BibNumber'>3</span>)
+ (<span class='BibYear'>2005</span>),
+ <span class='BibPages'>633–650</span>.
+</p>
```

Comment:
Why did this entry move around? Did anything about it change? If not, does that mean an unstable sorting is used, and so we can expect more "random" shuffling of entries? That'd be annoying. I'd expect entries here to be sorted by their MR number...?

Reply:
TBH I was not bothered to check - I just want to update Doc/Bib quickly as usual, and avoid having link for 2020 leading to an empty page. 

The sorting it relies on the order in gap-publishednicer.bib: 
https://github.com/gap-system/GapWWW/blob/0696138493507c784a9eab0c266fc4fc406aacd8/Doc/Bib/bibsort.g#L7-L9

One can't sort them by MR number, since it also contains various non-MR stuff. The order there is alphabetic by the surname of the 1st author. The sorting is done at 

https://github.com/gap-system/GapWWW/blob/0696138493507c784a9eab0c266fc4fc406aacd8/Doc/Bib/convbib.g#L95-L98

and `SIMPLE_STRING` was not changed for years. What I did differently from GAPDays 2020 though is that I run GAPDoc version from GitHub after @frankluebeck's comment at https://github.com/gap-system/GapWWW/issues/112. Let me try now with the latest GAPDoc release...

Reply:
And that does not change anything. Another thing that changed was that I have retrieved new raw data i.e. updated BibTeX records from MathSciNet, and that may have caused some shake-ups. So I suggest to change this, like before, and try to address this differently when there will be a major overhaul of GAP bibliography, not now.

Reply:
Actually, if I just run `convbib.g` now, it already produces a big diff -- without me updating any .bib files or anything else!

I also note that the sorting function in `convbib.g` only sorts by author, and doesn't sort stably -- so if two papers have the same authors, their relative order is not uniquely defined. In the past, we used to have stable sort by default, but not anymore... though I think that changed in GAP 4.9 (?) so already a few years back, and I guess you update more often than that?

Still, would help if we at least used `StableSort` instead of `Sort`. Even better would be if we sorted by authors *and* title. Something like this should do it:
`StableSortBy(bib[1], a -> [SIMPLE_STRING(a.author), a.title]);`

Of course the titles could also be passed through `SIMPLE_STRING`...

Ah and that's another issue I guess: `SIMPLE_STRING` seems to assume a specific one-byte encoding, perhaps ISO LATIN-1;  but I am pretty sure we are now getting UTF-8 data. So if any names contain non-ASCII chars, the ordering would also have changed.

All in all, this could is really brittle. It is indeed tempting to discard it all, and replace this Perl-GAP mix by something saner, written in a unicode aware language such as Python 3.

Reply:
There is of course even a bibtex parser for Python: https://bibtexparser.readthedocs.io/en/latest/

Anyway, let's merge this for now